sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2732,6 +2732,21 @@ object GpuOverrides extends Logging {
         override def convertToGpu(): GpuExpression =
           GpuCreateArray(childExprs.map(_.convertToGpu()), wrapped.useStringTypeWhenEmpty)
       }),
+    expr[ArrayDistinct](
+      "Removes duplicate values from the array",
+      ExprChecks.unaryProject(
+        TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 + TypeSig.NULL + 
+          TypeSig.ARRAY + TypeSig.STRUCT),
+        TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 + TypeSig.NULL + 
+          TypeSig.ARRAY + TypeSig.STRUCT),
+        TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 + TypeSig.NULL + 
+          TypeSig.ARRAY + TypeSig.STRUCT),
+        TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 + TypeSig.NULL + 
+          TypeSig.ARRAY + TypeSig.STRUCT)),
+      (a, conf, p, r) => new UnaryExprMeta[ArrayDistinct](a, conf, p, r) {
+        override def convertToGpu(child: Expression): GpuExpression =
+          GpuArrayDistinct(child)
+      }),
```

Comment:
Anonymous classes are unnecessarily difficult to debug when the conversion code throws #10838 

Look at https://github.com/NVIDIA/spark-rapids/pull/10839/files for an example